In [25]:
import pandas as pd
import numpy as np

# 1.2 Cargar el archivo con el separador ;
df = pd.read_csv('Cars.csv', sep=';')

# 1.3 y 1.4 Primer vistazo
print("Primeros 5 registros")
display(df.head(5))

print("Últimos 5 registros")
display(df.tail(5))

# 1.5 y 1.6 Dimensiones e Info
print(f"Dimensiones: {df.shape}")
print("Información General del Dataset")
df.info()

Primeros 5 registros


Numero                     Modelo Categoria   MPG  Cilindros  \
0       1  chevrolet chevelle malibu         D  18.0          8   
1       2          buick skylark 320         F  15.0          8   
2       3         plymouth satellite         A  18.0          8   
3       4              amc rebel sst         A  16.0          8   
4       5                ford torino         D  17.0          8   

   Desplazamiento  Fuerza    Peso  Aceleracion  Ano Origen  Choques  Taller  \
0           307.0   130.0  3504.0         12.0   70     US      4.0       4   
1           350.0   165.0  3693.0         11.5   70     US      3.0      10   
2           318.0   150.0  3436.0         11.0   70     US      1.0       4   
3           304.0   150.0  3433.0         12.0   70     US      2.0      10   
4           302.0   140.0  3449.0         10.5   70     US      3.0       3   

   Propietarios  
0             4  
1             6  
2             3  
3             2  
4             3

Últimos 5 registros


Numero           Modelo Categoria   MPG  Cilindros  Desplazamiento  \
389     390  ford mustang gl         C  27.0          4           140.0   
390     391        vw pickup         F  44.0          4            97.0   
391     392    dodge rampage         F  32.0          4           135.0   
392     393      ford ranger         C  28.0          4           120.0   
393     394       chevy s-10         G  31.0          4           119.0   

     Fuerza    Peso  Aceleracion  Ano  Origen  Choques  Taller  Propietarios  
389    86.0  2790.0         15.6   82      US      3.0       9             5  
390    52.0  2130.0         24.6   82  Europe      1.0       9             2  
391    84.0  2295.0         11.6   82      US      4.0      10             4  
392    79.0  2625.0         18.6   82      US      4.0       7             1  
393    82.0  2720.0         19.4   82      US      3.0       3             5

Dimensiones: (394, 14)
Información General del Dataset
<class 'pandas.DataFrame'>
RangeIndex: 394 entries, 0 to 393
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Numero          394 non-null    int64  
 1   Modelo          394 non-null    str    
 2   Categoria       390 non-null    str    
 3   MPG             390 non-null    float64
 4   Cilindros       394 non-null    int64  
 5   Desplazamiento  393 non-null    float64
 6   Fuerza          391 non-null    float64
 7   Peso            392 non-null    float64
 8   Aceleracion     394 non-null    float64
 9   Ano             394 non-null    int64  
 10  Origen          390 non-null    str    
 11  Choques         390 non-null    float64
 12  Taller          394 non-null    int64  
 13  Propietarios    394 non-null    int64  
dtypes: float64(6), int64(5), str(3)
memory usage: 43.2 KB


In [5]:
# 2.1 Verificar valores nulos por columna
print("Valores Faltantes")
print(df.isnull().sum())

# 2.2 Verificar si 'Fuerza' es un número o texto
print("Tipo de dato de la columna Fuerza")
print(df['Fuerza'].dtype)

Valores Faltantes
Numero            0
Modelo            0
Categoria         4
MPG               4
Cilindros         0
Desplazamiento    1
Fuerza            3
Peso              2
Aceleracion       0
Ano               0
Origen            4
Choques           4
Taller            0
Propietarios      0
dtype: int64
Tipo de dato de la columna Fuerza
float64


In [6]:
# Contar cuántas filas tenemos antes de borrar
filas_antes = len(df)

# Borrar todas las filas que tengan AL MENOS un valor vacío (NaN)
df = df.dropna()

# Contar cuántas filas quedaron
filas_despues = len(df)

print(f"Limpieza de nulos")
print(f"Filas eliminadas: {filas_antes - filas_despues}")
print(f"Filas restantes: {filas_despues}")

# Verificar que ya no queden nulos en ninguna columna
print("Conteo de nulos final:")
print(df.isnull().sum())

Limpieza de nulos
Filas eliminadas: 22
Filas restantes: 372
Conteo de nulos final:
Numero            0
Modelo            0
Categoria         0
MPG               0
Cilindros         0
Desplazamiento    0
Fuerza            0
Peso              0
Aceleracion       0
Ano               0
Origen            0
Choques           0
Taller            0
Propietarios      0
dtype: int64


In [27]:
# Contar cuántas filas están exactamente repetidas
n_duplicados = df.duplicated().sum()
print(f"Se encontraron {n_duplicados} registros duplicados.")

Se encontraron 0 registros duplicados.


In [8]:
# 3.1 Obtener el resumen estadístico completo
print("Resumen Estadístico del Dataset")
display(df.describe())

Resumen Estadístico del Dataset


Numero         MPG   Cilindros  Desplazamiento      Fuerza  \
count  372.000000  372.000000  372.000000      372.000000  372.000000   
mean   198.741935   23.537366    5.432796      192.689516  103.989247   
std    113.680301    7.718789    1.699310      103.956325   38.220595   
min      1.000000    9.000000    3.000000       70.000000   46.000000   
25%    100.750000   17.500000    4.000000      105.000000   75.000000   
50%    197.500000   23.000000    4.000000      145.500000   92.000000   
75%    298.250000   29.000000    8.000000      260.500000  120.500000   
max    394.000000   44.600000    8.000000      455.000000  230.000000   

              Peso  Aceleracion         Ano     Choques      Taller  \
count   372.000000   372.000000  372.000000  372.000000  372.000000   
mean   2966.048387    15.547043   76.048387    2.548387    6.475806   
std     849.158317     2.729586    3.686915    1.795569    2.226272   
min    1613.000000     8.000000   70.000000    0.000000    3.000000   
25%    2225.250000    13.700000   73.000000    1.000000    5.000000   
50%    2781.500000    15.500000   76.000000    3.000000    6.000000   
75%    3571.000000    17.025000   79.000000    4.000000    8.000000   
max    5140.000000    24.800000   82.000000    5.000000   10.000000   

       Propietarios  
count    372.000000  
mean       3.521505  
std        1.735027  
min        1.000000  
25%        2.000000  
50%        4.000000  
75%        5.000000  
max        6.000000

In [9]:
# 3.2 Calculando medidas específicas para la columna Fuerza
print(f"Media de Fuerza: {df['Fuerza'].mean():.2f}")
print(f"Mediana de Fuerza: {df['Fuerza'].median():.2f}")
print(f"Moda de Fuerza: {df['Fuerza'].mode()[0]}")

Media de Fuerza: 103.99
Mediana de Fuerza: 92.00
Moda de Fuerza: 88.0


In [10]:
# 3.3 Matriz de correlación entre las variables clave
columnas_interes = ['Peso', 'Fuerza', 'Cilindros', 'MPG']
matriz = df[columnas_interes].corr()

print("Matriz de Correlación")
display(matriz)

Matriz de Correlación


Peso    Fuerza  Cilindros       MPG
Peso       1.000000  0.860207   0.897497 -0.833412
Fuerza     0.860207  1.000000   0.839632 -0.776255
Cilindros  0.897497  0.839632   1.000000 -0.778485
MPG       -0.833412 -0.776255  -0.778485  1.000000

In [13]:
# 4.1 Seleccionar solo algunas columnas específicas
df_reducido = df[['Modelo', 'Fuerza', 'Origen']]
print("Vista del dataset reducido")
display(df_reducido.head())

Vista del dataset reducido


Modelo  Fuerza Origen
0  chevrolet chevelle malibu   130.0     US
1          buick skylark 320   165.0     US
2         plymouth satellite   150.0     US
3              amc rebel sst   150.0     US
4                ford torino   140.0     US

In [12]:
# 4.2 Filtrar autos con Fuerza mayor a 200
autos_potentes = df[df['Fuerza'] > 200]
print(f"Se encontraron {len(autos_potentes)} autos con más de 200 de fuerza.")
display(autos_potentes)

Se encontraron 10 autos con más de 200 de fuerza.


Numero                        Modelo Categoria   MPG  Cilindros  \
6         7              chevrolet impala         C  14.0          8   
7         8             plymouth fury iii         I  14.0          8   
8         9              pontiac catalina         C  14.0          8   
13       14       buick estate wagon (sw)         I  14.0          8   
25       26                     ford f250         C  10.0          8   
27       28                    dodge d200         D  11.0          8   
66       67               mercury marquis         A  11.0          8   
93       94  chrysler new yorker brougham         G  13.0          8   
94       95      buick electra 225 custom         C  12.0          8   
115     116            pontiac grand prix         G  16.0          8   

     Desplazamiento  Fuerza    Peso  Aceleracion  Ano Origen  Choques  Taller  \
6             454.0   220.0  4354.0          9.0   70     US      0.0       5   
7             440.0   215.0  4312.0          8.5   70     US      4.0       5   
8             455.0   225.0  4425.0         10.0   70     US      5.0       8   
13            455.0   225.0  3086.0         10.0   70     US      0.0       8   
25            360.0   215.0  4615.0         14.0   70     US      4.0      10   
27            318.0   210.0  4382.0         13.5   70     US      2.0       9   
66            429.0   208.0  4633.0         11.0   72     US      0.0       4   
93            440.0   215.0  4735.0         11.0   73     US      2.0       8   
94            455.0   225.0  4951.0         11.0   73     US      0.0       3   
115           400.0   230.0  4278.0          9.5   73     US      1.0       4   

     Propietarios  
6               2  
7               2  
8               5  
13              4  
25              1  
27              6  
66              3  
93              5  
94              1  
115             5

In [21]:
# 4.3 Filtro de autos del año 70 Y con fuerza > 200
filtro_pro = df[(df['Ano'] == 70) & (df['Fuerza'] > 200)]

print("Autos más potentes del año 70")
display(filtro_pro)

Autos más potentes del año 70


Numero                   Modelo Categoria   MPG  Cilindros  \
6        7         chevrolet impala         C  14.0          8   
7        8        plymouth fury iii         I  14.0          8   
8        9         pontiac catalina         C  14.0          8   
13      14  buick estate wagon (sw)         I  14.0          8   
25      26                ford f250         C  10.0          8   
27      28               dodge d200         D  11.0          8   

    Desplazamiento  Fuerza    Peso  Aceleracion  Ano Origen  Choques  Taller  \
6            454.0   220.0  4354.0          9.0   70     US      0.0       5   
7            440.0   215.0  4312.0          8.5   70     US      4.0       5   
8            455.0   225.0  4425.0         10.0   70     US      5.0       8   
13           455.0   225.0  3086.0         10.0   70     US      0.0       8   
25           360.0   215.0  4615.0         14.0   70     US      4.0      10   
27           318.0   210.0  4382.0         13.5   70     US      2.0       9   

    Propietarios  
6              2  
7              2  
8              5  
13             4  
25             1  
27             6

In [18]:
# 4.4 Filtrar solo autos de origen Japonés
autos_japoneses = df[df['Origen'] == 'Japan']
print(f"Cantidad de autos japoneses: {len(autos_japoneses)}")

Cantidad de autos japoneses: 76


In [19]:
# 5.1 Conteo de autos por Origen
print("Cantidad de autos por Origen")
print(df['Origen'].value_counts())

# Conteo de autos por Categoría
print("Cantidad de autos por Categoría")
print(df['Categoria'].value_counts())

Cantidad de autos por Origen
Origen
US        233
Japan      76
Europe     63
Name: count, dtype: int64
Cantidad de autos por Categoría
Categoria
I    54
H    48
F    47
G    47
C    41
A    39
D    34
B    33
E    29
Name: count, dtype: int64


In [20]:
# 5.2 Agrupar por Origen y calcular promedios de MPG y Fuerza
agrupado_origen = df.groupby('Origen')[['MPG', 'Fuerza']].mean()

print("Promedios de Rendimiento por Origen")
display(agrupado_origen)

# Agrupar por Categoría para ver el Peso promedio
agrupado_cat = df.groupby('Categoria')[['Peso']].mean()
print("Peso promedio por Categoría")
display(agrupado_cat)

Promedios de Rendimiento por Origen


MPG      Fuerza
Origen                       
Europe  27.661905   80.952381
Japan   30.381579   79.828947
US      20.189700  118.098712

Peso promedio por Categoría


Peso
Categoria             
A          3019.589744
B          2743.212121
C          3119.292683
D          2981.823529
E          2850.344828
F          2869.042553
G          2995.723404
H          3080.541667
I          2956.240741

Conclusiones
- Limpieza de Datos: Se identificaron y eliminaron filas con valores nulos, reduciendo el dataset de 394 a 372 registros, lo que garantiza cálculos precisos de promedios y tendencias.

- Integridad: Se confirmó que el dataset no contenía registros duplicados.

- Desempeño: Las estadísticas mostraron que variables como la Fuerza y el Peso son determinantes para entender el rendimiento de los autos, permitiendo comparar de forma clara los diferentes orígenes.

- Gracias a comandos como dropna() y drop_duplicates(), logramos depurar el dataset de manera rápida y eficiente, eliminando errores y repeticiones mediante código.

Herramientas usadas:
Claude y Gemini.
